io/eolearn/io/processing_api.py

```diff
@@ -98,64 +65,118 @@ def execute(self, eopatch=None, bbox=None, time_interval=None):
+        temporal_dim = len(timestamp) if timestamp else 1
+        shape = temporal_dim, size_y, size_x
+        self._extract_data(eopatch, images, shape)
 
-        shape = len(eopatch.timestamp), size_y, size_x
-
-        for f_type, f_name_src, f_name_dst in self.additional_data:
-            eopatch[(f_type, f_name_dst)] = self._extract_additional_data(images, f_type, f_name_src, shape)
-
-        if self.bands:
-            self._extract_bands(eopatch, images, shape)
-
-        eopatch.meta_info['service_type'] = 'processing'
         eopatch.meta_info['size_x'] = size_x
         eopatch.meta_info['size_y'] = size_y
-        eopatch.meta_info['maxcc'] = self.maxcc
         eopatch.meta_info['time_interval'] = time_interval
-        eopatch.meta_info['time_difference'] = self.time_difference
+        eopatch.meta_info['service_type'] = 'processing'
+
+        self._add_meta_info(eopatch)
 
         return eopatch
 
-    def _request_payload(self, date, bbox, size_x, size_y):
-        """ Build the payload dictionary for the request
+    def _extract_data(self, eopatch, images, shape):
+        """ Extract data from the received images and assign them to eopatch features
         """
-        date_from, date_to = date - self.time_difference, date + self.time_difference
-        time_from, time_to = date_from.isoformat() + 'Z', date_to.isoformat() + 'Z'
+        raise NotImplementedError("The _extract_data method should be implemented by the subclass.")
 
-        responses = [shr.response('default', MimeType.TIFF.get_string()), shr.response('userdata', 'application/json')]
+    def _build_payloads(self, bbox, size_x, size_y, timestamp):
+        """ Build payloads for the requests to the service
+        """
+        raise NotImplementedError("The _build_payloads method should be implemented by the subclass.")
 
-        data = shr.data(time_from=time_from, time_to=time_to, data_type=self.data_source.api_identifier())
-        data['dataFilter']['maxCloudCoverage'] = int(self.maxcc * 100)
+    def _get_timestamp(self, time_interval, bbox):
+        """ Get the timestamp array needed as a parameter for downloading the images
+        """
 
-        return shr.body(
-            request_bounds=shr.bounds(crs=bbox.crs.opengis_string, bbox=list(bbox)),
-            request_data=[data],
-            request_output=shr.output(size_x=size_x, size_y=size_y, responses=responses),
-            evalscript=self.generate_evalscript()
+    def _add_meta_info(self, eopatch):
+        """ Add any additional meta data to the eopatch
+        """
+
+
+class SentinelHubInputTask(SentinelHubInputBase):
+    """ A processing API input task that loads 16bit integer data and converts it to a 32bit float feature.
+    """
+    def __init__(self, data_source, size=None, resolution=None, bands_feature=None, bands=None, additional_data=None,
+                 maxcc=1.0, time_difference=None, cache_folder=None, max_threads=None, config=None,
```

Comment:
Parameter `max_threads` is not passed forward to parent class. (PyCharm marks such things ;) )

Reply:
Done :)